corehq/apps/app_manager/util.py

```diff
@@ -134,9 +134,11 @@ def change_xmlns(xform, old_xmlns, new_xmlns):
     else:
         GENERIC_XMLNS = "http://www.w3.org/2002/xforms"
         tag_xmlns = xform.data_node.tag_xmlns
+        form_xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
         if not tag_xmlns or tag_xmlns == GENERIC_XMLNS:  # no xmlns
-            xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
-            xml = change_xmlns(xform, GENERIC_XMLNS, xmlns)
+            xml = change_xmlns(xform, GENERIC_XMLNS, form_xmlns)
+        elif tag_xmlns != form_xmlns:
+            xml = change_xmlns(xform, tag_xmlns, form_xmlns)
```

Comment:
@emord I thought xmlns was derived from `form.unique_id` because of this code: https://github.com/dimagi/commcare-hq/blob/46c4645394eae4474a21f00a7028fed8c95273b5/corehq/apps/app_manager/util.py#L124-L125

That's right, Vellum does generate unique IDs too, although it's always conditional:

- https://github.com/dimagi/Vellum/blob/747d5b9c2d4054563f831766ad0b0f503e079bc5/src/parser.js#L196-L198
- https://github.com/dimagi/Vellum/blob/747d5b9c2d4054563f831766ad0b0f503e079bc5/src/writer.js#L83-L86

I get so confused trying to find a source of truth in HQ.